lib/BackbeatConsumer.js

```diff
@@ -542,6 +547,68 @@ class BackbeatConsumer extends EventEmitter {
         return subscriptions.length > 0;
     }
 
+    /**
+     * Helper method to wait for consumers to be assigned to partitions before
+     * proceeding to send canary messages.
+     * @param {number} wait - current accumulated wait time
+     * @param {callback} cb - callback to invoke
+     * @return {undefined}
+     */
+    _waitForAssignment(wait, cb) {
+        setTimeout(() => {
+            const assignments = this._consumer.assignments();
+            if (assignments.length === 0) {
+                if (wait > 50000) {
+                    return cb(true);
+                }
+                return this._waitForAssignment(wait + 2000, cb);
+            }
+            return cb();
+        }, 2000);
+    }
+
+    /**
+     * On start, send canary (noop) messages to set partition offsets for this
+     * consumer group. Necessary to avoid pause/resume bug where new locations
+     * on pause will not be able to queue entries as no current offset to
+     * determine number of entries (lag) to consume on resume.
+     * @param {callback} cb - callback to invoke
+     * @return {undefined}
+     */
+    _sendCanary(cb) {
+        const crrTopic = withTopicPrefix(replicationTopic);
+        if (this._topic !== crrTopic) {
+            return process.nextTick(cb);
+        }
+        return this._waitForAssignment(0, err => {
+            if (err) {
+                this._log.debug('could not send canary on consumer init', {
```

Comment:
Is this a normal situation? If not, can you make it an error log instead?

Reply:
I'm not sure how to best handle these situations. I added this debug log in case assignment takes too long. The 50 sec timeout is not a calculated estimate or anything, but it seems like consumers will attach to partitions fairly quickly normally. Maybe around 10-20 secs is average i'd say. 

If it doesn't assign in 50 sec, honestly probably something wrong internally with kafka and wouldn't reach this phase or fail here probably? : https://github.com/scality/backbeat/blob/4781125a0c89db2280c1ae32971686bbcdaabba4/lib/BackbeatConsumer.js#L244